### PR TITLE
Implement code connect for segmentedControl and heading

### PR DESCRIPTION
**Pull request by lukasoppermann**
Is this the correct code output? 

CleanShot 2024-08-14 at 11 28 22@2x

Also @joshblack I think the SegmentedContorl has the same issue with leadingIcon where it does only accept a function but not an element. Can we fix that?

### CODE REVIEW

**Comment by joshblack** [comment]
> Also @joshblack I think the SegmentedContorl has the same issue with leadingIcon where it does only accept a function but not an element. Can we fix that?

@lukasoppermann definitely! I think it should be just as doable as the last component with this issue 👍 This could be a good issue for FR to take on (adding support for element to loadingIcon) since it should be quick

**Comment by lukasoppermann** [comment]
After discussing with @siddharthkp we should keep the current behavior in the segmentedControl, as it otherwise leaves us open to unwanted customization in the icons.

I talked to Figma and they are working on a fix. We are hoping to get this in the next week, but until then this is blocked.

**Review by langermank — CHANGES_REQUESTED** [review]

File: packages/react/package.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
     "@oddbird/popover-polyfill": "^0.4.4",
     "@primer/behaviors": "^1.7.2",
     "@primer/live-region-element": "^0.7.1",
-    "@primer/octicons-react": "^19.9.0",
+    "@primer/octicons-react": "^19.12.0",
```
[the remark this answers is not in the document]
Comment (langermank) [reply packages/react/package.json:87]: Could just try and get this passing and ready for review https://github.com/primer/react/pull/5422

[the remark this answers is not in the document]
Comment (lukasoppermann) [reply packages/react/package.json]: I did revert octicons but the snapshots don't go away.

@TylerJDev I do need this update, but we can do this in a separate PR.

**Review by langermank — APPROVED** [review]
Good to go 🚀

**Pull request merged**